Update publish.yaml to publish to pypi.org and fixes to Readme based on comments from the team

misraturp opened the pull request:
- Added workflow to publish to pypi.org. Previously only published to test.pypi.org
- Updates to setup.py to avoid haystack-ai installation erroring out during package installation
- Updates to the readme for clearer documentation and fixing code examples

misraturp committed:
Update publish.yaml to publish to pypi.org

Add workflow to publish to pypi

oconnoob commented:
@misraturp does this PR account for the changes to the README etc.? just dont want to publish the wrong version

misraturp committed:
Update transcriber.py

assemblyai_import was part of lazy import. Lazy import was removed before. Removing this line too.

misraturp committed:
Update README.md

Update read me to include correct installation option.
Includes comments and fixes from Ryan and Patrick.

misraturp committed:
Update requirements.txt

Haystack removed some of their latest releases, there is no 0.137.0 version anymore. By their advice, removed the version limitation on the requirements.

misraturp committed:
Update setup.py

Haystack removed some of their latest releases, there is no 0.137.0 version anymore. By their advice, removed the version limitation on the requirements.

misraturp commented:
Yes, now it includes all the updates including the README. @oconnoob

oconnoob reviewed (approved):
LGTM! 🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀
